### DISCUSSION TITLE
关于 skyline 渲染引擎，你们有什么想法？

### DISCUSSION THREAD

**Discussion by LeeJim** [Tencent/tdesign-miniprogram#1646] [Ideas]
skyline 渲染引擎：https://developers.weixin.qq.com/miniprogram/dev/framework/runtime/skyline/introduction.html

你们有关注过吗？有相关诉求吗？说一下你的想法

**Comment by biangwua** [comment]
当然有啊！首先我来讲一下这两个月开发对于skyline的看法吧！ 我年前接触skyline 只能说这个渲染引擎的scroll-view非常好用 我在项目当中需要渲染很多子节点（类似于微博的文章之类的） 然后发现其中的相关渲染能力很强 并且惯性动画做得很好

其次就是worklet动画 让很多难以实现的动画（例如Tabs标签下划线跟随swiper滑动、评论区弹出层跟随手指滑动收起等等业务动画）都在这个渲染引擎下渲染出来的

再次，我本人因为一些特殊原因，一直没有真正的接触APP原生开发，在一年前接触过flutter开发IOS聊天应用，感觉良好，但是主要的开发工作是微信的mp，因此也一直期待着小程序代码能够转换出APP代码（apk和ipa）,现在出来了Donut框架，配合skyline我目前已经完成了一部分的原生动画的书写与测试，感觉这个搭配（Donut+skyline）在未来将会是一个不错的开发框架！（虽然目前bug贼多 毕竟这两个现在都是Beta 两个Beta加在一起 每次运行控制台都能见到熟悉的红色 当然这个一部分原因还是在于我没有更多的了解这两个框架的文档）

最后 回到正题 我最开始学习小程序开发那会儿 接触了很多组件库 最终综合UI主题、Bug多少以及功能丰富度选择了vant，后来发现了TDesign这个腾讯mp组件库，感觉这个更符合当前主流软件的视觉体验！无论是交互还是颜色的选择都很不错！但是考虑到skyline暂时取消了伪类以及CSS Variables，这将会给之前普遍采用webview开发模式的开发者们带来不小的挑战（至少在我的项目中 这两个还是用的比较多的） 

总的来说 单就目前Tdesign的交互和动画以及功能丰富度来讲 我还是很期待能够开发出skyline版本 以期能够完成更好的交互体验！

**Comment by linesoft2** [comment]
目前感觉微信小程序的性能尤其是启动速度还是比较差，接入Skyline后就好了很多，而且Skyline支持手势系统以及更强大的类似于Flutter Hero的共享元素动画，很希望未来微信小程序开发者能够把自己的小程序转入Skyline，会大幅度提升小程序的使用体验

**Comment by Logosww** [comment]
Skyline引擎的目前的开发迭代进度还是挺快的，在一些特性的支持上也开始逐步完善了。目前已经迭代到了1.0.0版本，支持了CSS Variable，感觉TDesign-miniprogram的Skyline适配计划应该提上日程了，为了更好的性能和用户体验。

**Reply by biangwua** [reply]
是这样的 一旦CSS变量支持 那么大部分的组件库特性也完备了 现在主要的痛点还是主要在于开发工具不支持热更新、基础库bug太多、编译速度慢这些问题上面。

**Reply by EOMZON** [reply]
热更新这个是不是现在还没解决啊，今天试了试感觉缺少热更新在处理样式看效果时很繁琐

**Comment by createlinux** [comment]
tdesign预计什么时候支持skyline呢？

**Comment by dday55** [comment]
今年中旬才考虑动手吗?这样25年都不一定能上呀
